### ISSUE TITLE
Allow to ignore `meta refresh` header

### ISSUE THREAD

**Issue by shtrom** [j0k3r/graby#198]
This really is a request for an addition to the supported Site Patterns syntax, but there doesn't seem to be much in the way of the definition over at @fivefilters. As Graby seems to be often used in conjunction with the ftr-site-config, I'm filing it here, but please feel free to redirect me to a more appropriate location,

Some sites, for example [LifeHacker Australia](https://www.lifehacker.com.au/2019/03/we-tried-mark-zuckerbergs-tricks-for-looking-taller-in-photos/) serve a page with the desired content, but with a `meta` header forcing a redirect to another page. This seems to allow them to detect non-JS browser, as it is in a `noscript` tag.

```html
	<noscript>
	  <meta http-equiv="refresh" content="0;url=/?nojs=1">
	</noscript>
```

In the case of Graby and Site Patterns, it would be good to have some directive allowing to explicitely disable the following of `meta refresh`. Alternatively, maybe disabling or removing `noscript` elements might be desirable, but I feel like the former solution would be more effective.

**Comment by fivefilters** [comment]
Hi @shtrom, I've encountered situations where following meta refresh is important, but also cases where it's used to redirect to a 'browser not supported' type page when in fact for the purposes of content extraction, the page does contain enough for us to process. So yes, a way to turn this off would be good.

At the moment we do this in Full-Text RSS in a hacky way, by using the find_string/replace_string directives to rewrite the `<meta>` tag, e.g. in https://github.com/fivefilters/ftr-site-config/blob/7baf9d42bd05e5c20a41241b3c6e1693c975f21a/cnn.com.txt

```
# Avoid redirecting to 'unsupported browser' page
find_string: <meta http-equiv="refresh"
replace_string: <meta norefresh
```

As for a shared definition of the site pattern syntax and directives, I think that's a good idea. We've got this at the moment, but it could do with an update: https://help.fivefilters.org/full-text-rss/site-patterns.html#pattern-format

**Comment by shtrom** [comment]
Yup, I generally go by this doc to write my site patterns, but yeah, there's nothing that can get an Issue or a PR recorded against (:

I just tried the following site-config as `lifehacker.com.au.txt`, but somehow, Graby (and f43.me) still get redirected.
```
# Generated by FiveFilters.org's web-based selection tool
# Place this file inside your site_config/custom/ folder
# Source: https://siteconfig.fivefilters.org/grab.php?url=https%3A%2F%2Fwww.lifehacker.com.au%2F2019%2F03%2Fwe-tried-mark-zuckerbergs-tricks-for-looking-taller-in-photos%2F

body: //div[contains(concat(' ',normalize-space(@class),' '),' main__content ')]
author: //div[@class='meta__author']/a

# Avoid redirecting to 'nojs' page                                              
find_string: <meta http-equiv="refresh"
replace_string: <meta norefresh

test_url: https://www.lifehacker.com.au/2019/03/we-tried-mark-zuckerbergs-tricks-for-looking-taller-in-photos/
```

**Comment by shtrom** [comment]
Looking at the parsing log over at https://f43.me/feed/test, it seems the redirect is honoured before any string replacement happens. Maybe this is actually a bona fide bug in Graby?

**Comment by fivefilters** [comment]
Don't think it's a bug, just a problem with the site config format and it being used to deal with these somewhat tricky situations. How and when it's parsed makes a difference to the results. After the code to look for meta refresh redirects was added, I had to go back in and add additional code to apply string replacement before looking for meta refresh to fix this problem. We didn't have that before. I think documenting this better and actually coming to an agreement as to how the site config file should be interpreted when it comes to these things might be something to consider. Happy to collaborate on that with you and @j0k3r and anyone else interested.
